jdaviz/configs/imviz/helper.py

```diff
@@ -524,20 +529,22 @@ def link_image_data(app, link_type='pixels', wcs_fallback_scheme='pixels', wcs_u
 
     # if the plugin exists, send a message so that the plugin's state is updated and spinner
     # is shown (the plugin will make a call back here)
-    if update_plugin and 'imviz-links-control' in [item['name'] for item in app.state.tray_items]:
+    if 'imviz-links-control' in [item['name'] for item in app.state.tray_items]:
         link_plugin = app.get_tray_item_from_name('imviz-links-control')
-        link_plugin.linking_in_progress = True
+        if update_plugin:
+            link_plugin.linking_in_progress = True
     else:
         link_plugin = None
 
-    # TODO: If different viewers have the same _marktags key, the key actually
-    #       points to the same Data table. Subsequent attempt to remove the
-    #       same key in this loop will emit warning.
-    # Clear any existing markers. Otherwise, re-linking will crash.
-    # Imviz can have multiple viewers open at the same time and each
-    # tracks their own markers.
-    for viewer in app._viewer_store.values():
-        viewer.reset_markers()
+    if link_type == app._link_type and wcs_use_affine == app._wcs_use_affine:
+        data_already_linked = [link.data2 for link in app.data_collection.external_links]
+    else:
+        for viewer in app._viewer_store.values():
+            if len(viewer._marktags):
+                raise ValueError(f"cannot change link_type (from '{app._link_type}' to "
+                                 f"'{link_type}') when markers are present. "
+                                 f" Clear markers with viewer.remove_markers() first")
```

Comment:
Are you sure you don't want `reset_markers`? `remove_markers` would only remove those added with the default marker names.